app/models/template_kind.rb

```diff
@@ -6,4 +6,12 @@ class TemplateKind < ActiveRecord::Base
   has_many :os_default_templates
   validates :name, :presence => true, :uniqueness => true
   scoped_search :on => :name
+
+  after_save :clear_cache!
```

Comment:
Not sure, but maybe add `if name.changed?` so it will clear cache on name changes?